etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/EtorDomainRegistration.java

```diff
@@ -126,9 +126,14 @@ DomainResponse handleDemographics(DomainRequest request) {
     DomainResponse handleOrders(DomainRequest request) {
         Order<?> orders;
 
+        String submissionId = request.getHeaders().get("RecordId");
+        if (submissionId == null) {
+            logger.logError("Missing required header: RecordId");
```

Comment:
Should we also set submissionId to a string that says null in this case? Not sure if it will save as null or empty

Reply:
I checked that it will return null if either RecordId is not present or the value is empty. Not sure if this is what you're referring to

Reply:
I thought we were going to write this submissionId eventually into the file, will it error if we pass a null value in for the String?  Or would it be better if we made the string "null" so that it is able to print to the file

Reply:
Good point. We need to check for null when trying to write to the file. I don't think we want to pass a "null" string as the file could be overwritten by different submissions and we wouldn't have certainty that the metadata is unique to a specific order anymore

Reply:
That's what I was thinking as well @halprin: check for null in SendOrderUseCase and skip if null

Reply:
Made the change in this PR

Reply:
When looking into `DomainRequest.java`, `getHeaders()` returns a Map<String, String>. Following this information, if we search for a key that doesn't exist, it will return null. If we search for a key that has an empty value, it will return empty. What are your thoughts on adding `|| submissionId.isEmpty()` to the if statement so it will check for null and for empty?

Reply:
That's a good point.  We should check for empty because that would be the same as if RS provided the header but didn't provide a value.  And if it is empty, we should probably set `submissionId` to `null` so we can make the subsequent logic in the usecase easier.

Reply:
I tested it with an empty value for the `RecordId` header and the `request.getHeaders().get("RecordId")` call returned null. I'll see if I can find the docs or the code to validate this assumption

Reply:
Oh, so maybe we don't need this the extra part of the `if` statement.